### ISSUE TITLE
[Feature Request] Considering One way roads for cars

### ISSUE THREAD

**Issue by MishalJariwala** [matejker/everystreet#13]
Hello, how difficult would it be to make this algorithm work for cars and consider that cars can only go a particular way on one-way roads?

P.S. Great job on solving the CPP problem.

**Comment by matejker** [comment]
Hi Mishal,

Yeah this is a really good question. In _practice_ it should be easy to change, you need to:
A) change the layer / type of the roads:
```python
CUSTOM_FILTER = (
    '["highway"]["area"!~"yes"]["highway"!~"bridleway|bus_guideway|bus_stop|construction|cycleway|elevator|footway|'
    'motorway|motorway_junction|motorway_link|escalator|proposed|construction|platform|raceway|rest_area|'
    'path|service"]["access"!~"customers|no|private"]["public_transport"!~"platform"]'
    '["fee"!~"yes"]["foot"!~"no"]["service"!~"drive-through|driveway|parking_aisle"]["toll"!~"yes"]'
)
```
OR easier option if you don't care to much about the actual types, OSMnx has an attribute `network_type` in [`graph_from_place()`](https://osmnx.readthedocs.io/en/stable/osmnx.html#osmnx.graph.graph_from_place) which can be set to `drive`.

B) I am doing a simplification by turning the underlying street directed network / graph into an undirected:
```python
location = "The Grange, Edinburgh, Scotland"
org_graph = ox.graph_from_place(location, custom_filter=CUSTOM_FILTER)

# Simplifying the original directed multi-graph to undirected, so we can go both ways in one way streets
graph = ox.utils_graph.get_undirected(org_graph)  # this needs to be removed
fig, ax = ox.plot_graph(graph, node_zorder=2, node_color="k", bgcolor="w")
```

In theory, I am not 100% sure if the algorithm will provide the optimal result, though. My gut feeling is yes and in [Edmonds & Johnson](https://web.eecs.umich.edu/~pettie/matching/Edmonds-Johnson-chinese-postman.pdf) paper they discuss it in chapter 5. and 6.

Finally, I appreciate your question - an interesting one for sure! You flag it as a _Feature Request_, I am bit reluctant to implement driving option as the project is meant to promote running and cycling over fossil burning alternatives :)

**Comment by matheusgomesms** [comment]
Hi all!

First of all, thanks for the lovely work!

I am using the code to generate a route to collect 360 imagery. Given the large distance, I'm unable to do it by running.

I made some tweaks to make it able to drive, but the final result still doesn't seem to consider the direction of the streets.

The tweaks I've made so far:

This one works fine, using `network_type='drive` and not using the undirected function.
```
org_graph = ox.graph.graph_from_place("MY AREA", network_type='drive', simplify=True, retain_all=True, truncate_by_edge=False, which_result=None, buffer_dist=None, clean_periphery=None, custom_filter=None)
fig, ax = ox.plot_graph(org_graph, node_zorder=2, node_color="k", bgcolor="w")
```

Then, I do this, to remove the ways not connected to the network:

graph = ox.utils_graph.get_largest_component(org_graph,strongly=False)
fig, ax = ox.plot_graph(graph, node_zorder=2, node_color="k", bgcolor="w")

Then I do the rest as the original code, plot gpx, but then, the gpx shows that the graphs seems not directed.

Am I doing something wrong here?

(If I manage to fix it, I'll post full result here, obviously!)
